src/python/pants/backend/python/tasks/python_binary_create.py

```diff
@@ -38,6 +38,16 @@ def register_options(cls, register):
                   "created and the command line used to create it. This information may be helpful to you, but means "
                   "that the generated PEX will not be reproducible; that is, future runs of `./pants binary` will not "
                   "create the same byte-for-byte identical .pex files.")
+    # TODO: make an analogy to cls.register_jvm_tool that can be overridden for python subsystems
+    # by a python_requirement_library() target, not just via pants.ini!
+    register('--ipex-injected-pex-version', advanced=True, default='1.6.12', fingerprint=True,
```

Comment:
If you're going to pin the pex version, then above you should _not_ grab an arbitrary one specified by a user silently:
```python
    pex_dist = self._distributions.get('pex')
    if pex_dist:
      pex_reqs = pex_dist.requires()
    if not pex_dist:
      # If pex was not already resolved to a dist, add it.
      pex_reqs = [self._pex_requirement]
```

For sanity sake, the version of pex used here in pants code should be the version pants depends on. I think you can kill the option, and above never use any user-specified version. It's probably fine to just use `pex.version.__version__` even though that's not really a proper API yet. As soon as that API breaks, the Pants CI will break, so its safe.